servers/rendering/rendering_device.cpp

```diff
@@ -4365,37 +4413,67 @@ void RenderingDevice::draw_list_draw(DrawListID p_list, bool p_use_indices, uint
 		}
 	}
 #endif
+	RDD::UniformSetID *valid_descriptor_ids = (RDD::UniformSetID *)alloca(dl->state.set_count * sizeof(RDD::UniformSetID));
```

Comment:
In the `RenderingDevice` code we use a thread-local vector for this (not reiterating the comment on every case in the PR):

```suggestion
	thread_local LocalVector<RDD::UniformSetID> valid_descriptor_ids;
	valid_descriptor_ids.resize(dl->state.set_count);
```